Charlie port to v3

BenJamesBen opened the pull request:
I am seeing strange issues with Charlie, that I thought was related to dependency issues. However, now I am not so sure.

Sometimes when i load a fresh Charlie pattern, it incorrectly looks like this:
![Screen Shot 2022-09-04 at 7 52 04 PM](https://user-images.githubusercontent.com/109869956/188352593-72c5e6a7-32c6-4ea4-9ec4-bbbda0faa3b7.png)

But, then I refresh the web page, the pattern changes to something that looks more correct:
![Screen Shot 2022-09-04 at 7 52 22 PM](https://user-images.githubusercontent.com/109869956/188352598-433369c0-9795-4aad-8efa-1f7692e7ffbc.png)

And, when I look in the Events log, I see measurements like this:
![Screen Shot 2022-09-04 at 7 57 07 PM](https://user-images.githubusercontent.com/109869956/188352605-3efcf8a7-f2a6-41ed-90e7-54658a7fd18c.png)

But, when I refresh the web page, the numbers change to:
![Screen Shot 2022-09-04 at 7 57 17 PM](https://user-images.githubusercontent.com/109869956/188352616-80631c3b-468a-4515-b1d7-7092e4c9f7c8.png)

This is all using the same measurements. 

When the bug state occurs and I refresh the page to get the correct pattern, when I switch to the Events page and then back to the DraftDesign page, the pattern reverts back to the initial bad pattern.

joostdecock commented:
These are bugs in the lab. I suspect they were introduced when Enoch was working on the layout.

Essentially, it doesn't reset the state when you switch to a new design. Reloading the page will force the reset.

BenJamesBen commented:
Changing from draft to actual PR. I assume that because one of the pattern states is the correct pattern, that the Charlie port is correct.

I removed the redundant `after:`s that I initially added in `back.mjs` for investigation. (Specifically, I removed titanBack and titanFront. The back already has `from: titanBack` and `after: front` and the front is already `from: titanFront`. So, titanBack and titanFront aren't needed in back's `after:`.)

I filed #2736 to track the lab pattern state issue.
